content/post/108-js-ipfs-0.50.0.md

```diff
@@ -0,0 +1,201 @@
+
+These types of arrays hold numbers with an integer value range of 0-255 and support an API very similar to that of Node.js Buffers, which should come as no surprise as since v3.0.0 of Node.js, Buffers have been a subclass of Uint8Arrays.
+
+These are exciting times so going forwards as JavaScript becomes more capable and the browser support arrives we are reducing our dependence on core Node.js libraries and utilities. Part of that is removing all use of Node.js Buffers in our codebase.
+
+With js-ipfs@0.50.0 you should stop relying on Node.js Buffers to be returned from any part of the [Core-API], instead you should code against the Uint8Array interface.
+
+Some modules that we depend on will still return Buffers, which we pass on to avoid any conversion cost but we hope to remove or refactor these over time.  In order to remain forwards-compatible you should not use Node.js Buffer methods on any of these returned values.
+
+For example in the code below we create a Buffer from the string 'Hello', add it to IPFS then immediately cat it and call `toString()` on the chunks. This takes advantage of the fact that the Buffer we've added is `utf8` encoded.  Buffer.toString() takes an encoding argument which is `utf8` by default, so the below code works but it's only by coincidence:
+
+```javascript
+const { cid } = await ipfs.add(Buffer.from('Hello'))
+
+for await (const chunk of ipfs.cat(cid)) {
+  console.info(chunk.toString()) // prints 'Hello'
+}
+```
+
+Instead we would use the [TextEncoder](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder) and [TextDecoder](https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder) classes. These are explicit in the encoding/decoding they use (also `utf8` by default) so are safer to use:
+
+```javascript
+const encoder = new TextEncoder()
+const decoder = new TextDecoder()
+
+const { cid } = await ipfs.add(encoder.encode('Hello'))
+
+for await (const chunk of ipfs.cat(cid)) {
+  console.info(decoder.decode(chunk)) // prints 'Hello'
+}
+```
+
+# ✨New features
+
+* Store pins in datastore instead of a DAG ([#2771](https://github.com/ipfs/js-ipfs/issues/2771)) ([64b7fe4](https://github.com/ipfs/js-ipfs/commit/64b7fe41738cbe96d5a9075f0c01156c6f889c40))
+* Add protocol list to ipfs id output ([#3250](https://github.com/ipfs/js-ipfs/issues/3250)) ([1b6cf60](https://github.com/ipfs/js-ipfs/commit/1b6cf600a6b1348199457ca1fe6f314b6eff8c46))
+* IPNS publish in browser example ([#3207](https://github.com/ipfs/js-ipfs/issues/3207)) ([91faec6](https://github.com/ipfs/js-ipfs/commit/91faec6e3d89b0d9883b8d7815c276d44048e739))
+* Update hapi to v20 ([#3245](https://github.com/ipfs/js-ipfs/issues/3245)) ([1aeef89](https://github.com/ipfs/js-ipfs/commit/1aeef89c73f42a2f6cceb7f0598400141ce40e23))
+* Update to libp2p@0.29.0 ([63d4d35](https://github.com/ipfs/js-ipfs/commit/63d4d353c606e4fd487811d8a0014bb2173f11be))
+
+# 🏗 API Changes
+
+## Core API & HTTP API Client
```

Comment:
If these are backwards incompatible changes and require developers to update their code, should call that very clearly. A big shouty banner or whatnot. Something un-ignorable.

Reply:
Done!